Library/Homebrew/diagnostic.rb

```diff
@@ -443,6 +443,32 @@ def check_access_cellar
         EOS
       end
 
+      def check_for_empty_taps
+        taps_dir = HOMEBREW_LIBRARY/"Taps"
+        return unless taps_dir.exist?
+        return unless taps_dir.directory?
+
+        found_empty_dirs = []
+
+        # Warn USER of empty taps_dir children directories
+        taps_dir.children.select{ |c| c.directory? } \
```

Comment:
Another note, running `find . -empty -type d` to find all the empty directories in the Taps folder reveals that some Taps leave empty directories themselves, so just deleting all the empty directories recursively won't solve the problem either. However, limiting the depth of the deletion to two levels might solve both problems, which we can do with `find $(brew --prefix)/Library/Taps -empty -type d -depth 2 | xargs rmdir`.